ctapipe/image/muon/tests/test_features.py

```diff
@@ -14,6 +19,48 @@ def test_ring_containment():
 
     assert(ring_cont <= 1. and ring_cont >= 0.)
 
+
+def test_ring_completeness():
+
+    angle_ring = np.linspace(0, 2 * mt.pi, 360.)
+    x = np.cos(angle_ring) * u.m
+    y = np.sin(angle_ring) * u.m
+    pe = np.random.uniform(0, 100, len(x))
+    ring_radius = 1. * u.m
+
+    ring_center_x = 0. * u.m
+    ring_center_y = 0. * u.m
+
+    ring_comp = ring_completeness(
+        x, y, pe, ring_radius,
+        ring_center_x, ring_center_y,
+        30, 30)
+
+    assert(ring_comp <= 1. and ring_comp >= 0.)
+
+
+def test_npix_above_threshold():
+    len_array = 100
+    thr = 5.
+    pix = np.random.uniform(0, 100, len_array)
+
+    npix = npix_above_threshold(pix, thr)
+
+    assert((npix >= 0) and (npix <= len_array))
```

Comment:
I would not use randomness for tests ... So I'd propose
```
pix = np.ones(len_array)
pix[[3, 7, 77]] = thr + 1

npix = npix_above_threshold(pix, thr)

assert npix == 3
```

You might want to test (i.e. document) corner cases: what if pix is an empty array? What result would you expect? zero? or some exception?

Reply:
any particular reason not to use randomness here?
the assertion is very well defined

Reply:
I think tests should be deterministic if they can. What if a user reports this test to fail on their machine? Hard to help this user.

Maybe it already helps when one sets the random seed, but I did not think about this in the first place.